rocAL/include/loaders/image/numpy_loader.h

```diff
@@ -0,0 +1,91 @@
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "circular_buffer.h"
+#include "commons.h"
+#include "image_read_and_decode.h"
+//
+// NumpyLoader runs an internal thread for loading an decoding of numpy arrays asynchronously
+// it uses a circular buffer to store decoded numpy arrays for the user
+class NumpyLoader : public LoaderModule {
+   public:
+    explicit NumpyLoader(void* dev_resources);
+    ~NumpyLoader() override;
+    LoaderModuleStatus load_next() override;
+    void initialize(ReaderConfig reader_cfg, DecoderConfig decoder_cfg, RocalMemType mem_type, unsigned batch_size, bool keep_orig_size = false) override;
+    void set_output(Tensor* output_image) override;
+    void set_random_bbox_data_reader(std::shared_ptr<RandomBBoxCrop_MetaDataReader> randombboxcrop_meta_data_reader) override;
+    size_t remaining_count() override;  // returns number of remaining items to be loaded
+    void reset() override;              // Resets the loader to load from the beginning of the media
+    Timing timing() override;
+    void start_loading() override;
+    LoaderModuleStatus set_cpu_affinity(cpu_set_t cpu_mask);
+    LoaderModuleStatus set_cpu_sched_policy(struct sched_param sched_policy);
+    void set_gpu_device_id(int device_id);
+    std::vector<std::string> get_id() override;
+    decoded_image_info get_decode_image_info() override;
+    crop_image_info get_crop_image_info() override;
+    void set_prefetch_queue_depth(size_t prefetch_queue_depth) override;
+    void shut_down() override;
+    void feed_external_input(const std::vector<std::string>& input_images_names, const std::vector<unsigned char*>& input_buffer,
+                             const std::vector<ROIxywh>& roi_xywh, unsigned int max_width, unsigned int max_height, unsigned int channels, ExternalSourceFileMode mode, bool eos) override {}
+
+   private:
+    bool is_out_of_data();
+    void de_init();
+    void stop_internal_thread();
+    LoaderModuleStatus update_output_image();
+    LoaderModuleStatus load_routine();
+    std::shared_ptr<Reader> _reader;
+    std::shared_ptr<RandomBBoxCrop_MetaDataReader> _randombboxcrop_meta_data_reader = nullptr;
+    Tensor* _output_tensor;
+    std::vector<std::string> _output_names;  //!< image name/ids that are stores in the _output_image
+    size_t _output_mem_size;
+    MetaDataBatch* _meta_data = nullptr;  //!< The output of the meta_data_graph,
+    std::vector<std::vector<float>> _bbox_coords;
```

Comment:
```suggestion
    MetaDataBatch* _meta_data = nullptr;  //!< The output of the meta_data_graph,
    std::vector<std::vector<float>> _bbox_coords;
```

The above 2 variables not used